aten/src/ATen/cuda/cub.cu

```diff
@@ -2,6 +2,8 @@
 #include <ATen/cuda/cub.cuh>
 #include <ATen/cuda/CUDAConfig.h>
 
+#include <thrust/iterator/transform_iterator.h>
```

Comment:
The Thrust and CUB functors like `thrust::equal_to` have been deprecated in favor of the -identical- standard ones, which live in `<cuda/std/functional>`
```suggestion
#include <cuda/std/functional>
#include <thrust/iterator/transform_iterator.h>
```